cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -620,6 +622,34 @@ runQueryPoolState (AnyConsensusModeParams cModeParams) network poolIds = do
   result <- executeQuery era cModeParams localNodeConnInfo qInMode
   obtainLedgerEraClassConstraints sbe writePoolState result
 
+-- | Query the local mempool state
+runQueryTxMempool
+  :: AnyConsensusModeParams
+  -> NetworkId
+  -> TxMempoolQuery
+  -> Maybe OutputFile
+  -> ExceptT ShelleyQueryCmdError IO ()
+runQueryTxMempool (AnyConsensusModeParams cModeParams) network query mOutFile = do
+  SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
+  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
+
+  localQuery <- case query of
+      TxMempoolQueryTxExists tx -> do
+        anyE@(AnyCardanoEra era) <- determineEra cModeParams localNodeConnInfo
+        let cMode = consensusModeOnly cModeParams
+        eInMode <- toEraInMode era cMode
+          & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
+        pure $ LocalTxMonitoringQueryTx $ TxIdInMode tx eInMode
+      TxMempoolQueryNextTx -> pure LocalTxMonitoringSendNextTx
+      TxMempoolQueryInfo -> pure LocalTxMonitoringMempoolInformation
+
+  result <- liftIO $ queryTxMonitoringLocal localNodeConnInfo localQuery
+  let renderedResult = encodePretty result
+  liftIO $ LBS.putStrLn renderedResult
```

Comment:
So here if there is no `OutputFile` specified, we then print to stdout. If there is an `OutputFile`, we only write JSON to disk.

Reply:
That seems straightforward, and I'll make the change, but I'm unsure why this is wanted

When would I want to do something like

```sh
cardano-cli query tx-mempool next-tx --out-file foo
```

instead of

```sh
cardano-cli query tx-mempool next-tx > foo
```

Reply:
What you are saying makes sense but my suggestion mirrors the current behaviour of other cli commands.